src/Core/ConfigManager.ts

```diff
@@ -0,0 +1,34 @@
+import * as Discord from 'discord.js';
+import configData from '../Config/Botconfig.json';
+
+export class ConfigManager {
+    private static instance: ConfigManager;
+    private client: Discord.Client;
+
+    private moderatorRole: Array<string>;
+    private administratorRole: Array<string>;
+
+    private constructor(client: Discord.Client) {
+        this.client = client;
+    }
+
+    public static getInstance(client: Discord.Client): ConfigManager {
+        if (!this.instance) {
+            this.instance = new ConfigManager(client);
+        }
+        return this.instance;
+    }
+
+    /**
+     * This function will load config from Botconfig.json
+     */
+    public loadConfig(): void {
+        this.moderatorRole = configData.roles.moderator;
+        this.administratorRole = configData.roles.administrator;
+    }
+

```

Comment:
What could be the utility of `saveConfig`? I don't completely get it. @ASmallCookie

Reply:
``saveConfig`` is a function which save changes made in the **botconfig.json** file.

I plan to use it that way : 
```ts
let configManager = ConfigManager.getInstance();
let config = configManager.getConfig();
configManager.saveConfig(...config, {logger.saveInFile = true, logger.saveFilename = "hello.log"});
````
It's just an example
@qcarpentier